server.go

```diff
@@ -630,14 +633,36 @@ func (srv *Server) disposeBuffer(w *response) {
 }
 
 func (srv *Server) serveDNS(w *response) {
+	dh, off, err := unpackMsgHdr(w.msg, 0)
+	if err != nil {
+		// Let client hang, they are sending crap; any reply can be used to amplify.
+		return
+	}
+
 	req := new(Msg)
-	err := req.Unpack(w.msg)
-	if err != nil { // Send a FormatError back
-		x := new(Msg)
-		x.SetRcodeFormatError(req)
-		w.WriteMsg(x)
+	req.setHdr(dh)
+
+	switch srv.MsgAcceptFunc(dh) {
+	case MsgAccept:
+
```

Comment:
Nit: These newlines between switch cases is inconsistent with elsewhere in the library and idiomatic go.